cdap-unit-test/src/test/java/co/cask/cdap/spark/app/SparkSecureStoreApp.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spark.app;
+
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.spark.AbstractSpark;
+import co.cask.cdap.api.spark.JavaSparkExecutionContext;
+import co.cask.cdap.api.spark.JavaSparkMain;
+
+import java.util.HashMap;
+
+/**
+ * A program used to test the secure store apis
+ */
+public class SparkSecureStoreApp extends AbstractApplication {
+  private static final String KEY = "key";
+  private static final String VALUE = "value";
+  private static final String NAMESPACE = "namespace";
+
+  @Override
+  public void configure() {
+    setName("TestSecureStoreApis");
+    setDescription("App to test Spark using secure store apis");
+    addSpark(new SparkSecureStoreProgram());
+  }
+
+  public static class SparkSecureStoreProgram extends AbstractSpark implements JavaSparkMain {
+    @Override
+    public void configure() {
+      setName("SparkSecureStoreProgram");
+      setDescription("Test Spark with Streams from other namespace");
+      setMainClass(SparkSecureStoreProgram.class);
+    }
+
+    @Override
+    public void run(JavaSparkExecutionContext sec) throws Exception {
+      sec.getAdmin().put(NAMESPACE, KEY, VALUE.getBytes(), "", new HashMap<String, String>());
+      sec.get(NAMESPACE, KEY);
+      sec.list(NAMESPACE);
```

Comment:
Need to add a spark closure usage. E.g. parallelize a collection, perform a map reduce and collect. Inside the map function, use the `SecureStore` returned by the `sec.getSecureStore()` call.